### PR TITLE
Fix ENCODE file format mapping for compound format strings

### LINKED ISSUE #16: Fix ENCODE file format mapping for compound format strings
## Summary

The ENCODE file format mapping in `get_file_format()` fails to resolve 181,837 ENCODE files (22.6%) because ENCODE's metadata TSV uses compound format strings like `bigBed narrowPeak`, `bed bed3+`, and `bigWig bed3+`. The lookup dictionary has entries for the simple base formats (`bigBed`, `bed`, `bigWig`) but not for these compound variants, so they return `None` and the files end up with no `file_format` in the database.

## Root cause

`get_file_format()` in `ontology_mappings.py` performs an exact-match lookup against `FILE_FORMAT_TO_EDAM`:

```python
return FILE_FORMAT_TO_EDAM.get(encode_format.lower())
```

ENCODE's "File format" TSV column uses a `<container> <subtype>` convention (e.g. `bigBed narrowPeak`, `bed bed3+`). Two compound keys (`bed narrowPeak`, `bed broadPeak`) happen to be in the dictionary and work, but all other compound forms fail silently.

Additionally, six simple format keys are missing entirely from the dictionary: `starch`, `tagalign`, `biginteract`, `csfasta`, `csqual`, and `h5ad`.

**Breakdown of unmapped files by derived format:**

| Should map to | Count | Top compound values |
|--------------|------:|---------------------|
| bigBed | 120,994 | `bigBed narrowPeak` (89K), `bigBed bed3+` (15K), `bigBed broadPeak` (7K) |
| BED | 50,772 | `bed bed3+` (24K), `bed idr_ranked_peak` (17K), `bed bed3` (5K) |
| bigWig | 4,938 | `bigWig bed3+` |
| Other missing | 5,133 | `starch` (2.4K), `tagAlign` (1.7K), `gff gff3` (713), `bigInteract` (252), `h5ad` (32) |

## Affected code

- `src/cfdb/services/ontology_mappings.py` — `FILE_FORMAT_TO_EDAM` dictionary and `get_file_format()` function

### PR DESCRIPTION
**Pull request by conradbzura**
## Summary

`get_file_format()` in `ontology_mappings.py` performs an exact-match lookup against `FILE_FORMAT_TO_EDAM`, but ENCODE's metadata TSV uses compound format strings like `bigBed narrowPeak` and `bed bed3+`. This causes 181,837 ENCODE files (22.6%) to have no `file_format` in the database.

Add a first-token fallback to `get_file_format()` so compound formats resolve to their base container format, normalize all dictionary keys to lowercase for consistent case-insensitive lookup, and add six missing simple format entries.

Closes #16

### 1. Add first-token fallback to `get_file_format()`

The function now tries an exact lookup on the lowercased input first. If that fails, it splits on whitespace and retries with just the first token. This handles all `<container> <subtype>` patterns (e.g. `bigBed narrowPeak` → `bigbed` → `bigBed`, `bed bed3+` → `bed` → `BED`).

```python
# Before
return FILE_FORMAT_TO_EDAM.get(encode_format.lower())

# After
key = encode_format.lower()
result = FILE_FORMAT_TO_EDAM.get(key)
if result:
    return result
base = key.split()[0]
return FILE_FORMAT_TO_EDAM.get(base)
```

### 2. Normalize dictionary keys to lowercase

Mixed-case keys like `bigBed`, `bigWig`, `broadPeak`, `narrowPeak`, `bedGraph`, and `CEL` are normalized to lowercase so the `.lower()` lookup works consistently. The EDAM `name` values retain their original casing.

### 3. Add missing simple format entries to `FILE_FORMAT_TO_EDAM`

Six format strings present in ENCODE's TSV have no dictionary entry at all:

| Key | EDAM ID | EDAM Name | Rationale |
|-----|---------|-----------|-----------| 
| `starch` | `format:3003` | BED | Compressed BED archive (BEDOPS) |
| `tagalign` | `format:3003` | BED | tagAlign is a BED variant |
| `biginteract` | `format:3004` | bigBed | bigInteract is a bigBed variant |
| `csfasta` | `format:1929` | FASTA | Color-space FASTA (SOLiD) |
| `csqual` | `format:2330` | Plain text | Color-space quality scores |
| `h5ad` | `format:3590` | HDF5 | AnnData HDF5 format |

## Test cases

| Test Suite | Test ID | Given | When | Then | Coverage Target |
|------------|---------|-------|------|------|-----------------|
| `TestGetFileFormat` | FF-001 | A simple format string `"fastq"` | `get_file_format("fastq")` is called | Returns `{"id": "format:1930", "name": "FASTQ"}` | Existing exact-match behavior |
| `TestGetFileFormat` | FF-002 | A compound format string `"bigBed narrowPeak"` | `get_file_format("bigBed narrowPeak")` is called | Returns `{"id": "format:3004", "name": "bigBed"}` | First-token fallback |
| `TestGetFileFormat` | FF-003 | A compound format string `"bed bed3+"` | `get_file_format("bed bed3+")` is called | Returns `{"id": "format:3003", "name": "BED"}` | First-token fallback for BED variants |
| `TestGetFileFormat` | FF-004 | A compound format string `"bigWig bed3+"` | `get_file_format("bigWig bed3+")` is called | Returns `{"id": "format:3006", "name": "bigWig"}` | First-token fallback for bigWig variants |
| `TestGetFileFormat` | FF-005 | An existing compound key `"bed narrowPeak"` | `get_file_format("bed narrowPeak")` is called | Returns `{"id": "format:3613", "name": "NarrowPeak"}` | Exact match takes precedence over fallback |
| `TestGetFileFormat` | FF-006 | A new simple format `"starch"` | `get_file_format("starch")` is called | Returns `{"id": "format:3003", "name": "BED"}` | New dictionary entry |
| `TestGetFileFormat` | FF-007 | A new simple format `"tagAlign"` | `get_file_format("tagAlign")` is called | Returns `{"id": "format:3003", "name": "BED"}` | New dictionary entry, case-insensitive |
| `TestGetFileFormat` | FF-008 | A new simple format `"bigInteract"` | `get_file_format("bigInteract")` is called | Returns `{"id": "format:3004", "name": "bigBed"}` | New dictionary entry |
| `TestGetFileFormat` | FF-009 | A new simple format `"h5ad"` | `get_file_format("h5ad")` is called | Returns `{"id": "format:3590", "name": "HDF5"}` | New dictionary entry |
| `TestGetFileFormat` | FF-010 | An empty string `""` | `get_file_format("")` is called | Returns `None` | Empty input guard |
| `TestGetFileFormat` | FF-011 | A completely unknown format `"xyzzy"` | `get_file_format("xyzzy")` is called | Returns `None` | Unknown format returns None |
| `TestGetFileFormat` | FF-012 | A compound string with unknown base `"xyzzy foo"` | `get_file_format("xyzzy foo")` is called | Returns `None` | Unknown compound format returns None |

## Implementation plan

1. - [x] Write tests for `get_file_format()` covering exact match, first-token fallback, precedence, new entries, and edge cases (FF-001 through FF-012)
2. - [x] Add missing format entries (`starch`, `tagalign`, `biginteract`, `csfasta`, `csqual`, `h5ad`) to `FILE_FORMAT_TO_EDAM`
3. - [x] Normalize mixed-case dictionary keys to lowercase for consistent `.lower()` lookup
4. - [x] Add first-token fallback logic to `get_file_format()`

### CODE REVIEW

File: tests/test_ontology_mappings.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    def test_get_file_format_with_simple_format(self):
+        """
+        GIVEN a simple format string that exactly matches a dictionary key
+        WHEN get_file_format is called with "fastq"
+        THEN the EDAM FASTQ term is returned
+        """
+        # Act
+        result = get_file_format("fastq")
+
+        # Assert
```
Comment (conradbzura) [inline tests/test_ontology_mappings.py]: All 12 test docstrings need to follow the format from testguide-py.md section 4. The first line must start with `Test`, followed by a blank line, then `Given:`, `When:`, `Then:` each on their own line with a colon, content indented 4 spaces, and `Then` starting with "It should...".

Expected format:
```python
"""Test exact match lookup for a simple format string.

Given:
    A simple format string "fastq" that exactly matches a dictionary key
When:
    get_file_format is called with "fastq"
Then:
    It should return the EDAM FASTQ term
"""
```

File: tests/test_ontology_mappings.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+"""Tests for ENCODE file format mapping in ontology_mappings."""
+
+from __future__ import annotations
+
+from cfdb.services.ontology_mappings import get_file_format
+
+
+class TestGetFileFormat:
```
Comment (conradbzura) [inline tests/test_ontology_mappings.py]: `get_file_format` is a module-level function, not a class. The test guide section 2 says "Tests for module-level functions are module-level test functions." Consider making these top-level `test_get_file_format_*` functions instead of wrapping them in a class.

File: tests/test_ontology_mappings.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        """
+        GIVEN an unrecognized compound format string "xyzzy foo"
+        WHEN get_file_format is called
+        THEN None is returned (neither the full string nor the first token matches)
+        """
+        # Act
+        result = get_file_format("xyzzy foo")
+
+        # Assert
+        assert result is None
```
Comment (conradbzura) [inline tests/test_ontology_mappings.py]: The test guide section 9 says Hypothesis must be used for "any function with a wide input domain." `get_file_format` accepts arbitrary strings, so consider adding a property test verifying the return-type invariant — for any string, the result is always `None` or a dict with `"id"` and `"name"` keys.
